wandb/sync/sync.py

```diff
@@ -94,7 +98,20 @@ def _parse_pb(self, data, exit_pb=None):
                 pb.run.entity = self._entity
             if self._job_type:
                 pb.run.job_type = self._job_type
-            pb.control.req_resp = True
+            if self._add_tag:
+                pb.run.tags.extend(self._add_tag)
+            if self._remove_tag:
+                for tag in self._remove_tag:
+                    if tag not in pb.run.tags:
+                        wandb.termwarn(
+                            f"Warning: Tag '{tag}' does not exist and cannot be removed."
+                        )
+                    else:
+                        pb.run.tags[:] = [
+                            existing_tag
+                            for existing_tag in pb.run.tags
+                            if existing_tag != tag
+                        ]
```

Comment:
The order of tags doesn't matter, right?

```suggestion
                tags_to_remove = set(self._remove_tag)
                tags_on_run = set(pb.run.tags)
                for tag in tags_to_remove - tags_on_run:
                    wandb.termwarn(f"Tag '{tag}' does not exist; ignoring.")
                pb.run.tags = list(tags_on_run - tags_to_remove)
```